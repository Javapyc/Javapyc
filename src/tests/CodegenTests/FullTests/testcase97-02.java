class Main { 
	// email me if you want the original joke. I'm Sriram
	public static void main (String[ ] ID) { 
		Dude minister = new Minister();
		Dude priest = new Priest();
		Rabbi rabbi = new Rabbi();

		if (minister.walkOnWater(true))
		{
			int i = rabbi.astonish();
		}
		else {}
		if (priest.walkOnWater(true))
		{
			int i = rabbi.astonish();
		}
		else {}
		boolean b = rabbi.walkOnWater(false);
		int i = priest.speculate();
	} 
}

class Dude {
	public boolean walkOnWater(boolean knowsWhereRocksAre) {
		if (!knowsWhereRocksAre) {
			int i = this.flail();
		}
		else {}
		return knowsWhereRocksAre;
	}

	public int flail() {
		System.out.println(13); // the universal number of flailing
		return 0;
	}

	public int speculate() {
		System.out.println(21); // the universal number of speculation
		return 0;
	}
}

class Minister extends Dude {}
class Priest extends Dude {}
class Rabbi extends Dude {
	public int astonish() {
		System.out.println(34); // the universal number of astonishment
		return 0;
	}
}

