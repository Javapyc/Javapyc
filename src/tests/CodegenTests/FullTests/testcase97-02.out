34
34
13
21
